datafusion/src/physical_plan/planner.rs

```diff
@@ -302,7 +302,7 @@ impl DefaultPhysicalPlanner {
         ctx_state: &ExecutionContextState,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let batch_size = ctx_state.config.batch_size;
-
+        let max_partitions = ctx_state.config.target_partitions;
```

Comment:
since this is only used within the `LogicalPlan::TableScan` match arm, it's better to move the assignment inside there.

Reply:
Got it.  fix `batch_size` too